File: TcpSocketSystem_DotNet/src/Domain/Puertos/IServicioLogging.cs
namespace TcpSocketSystem.Core.Domain.Puertos;

/// <summary>
/// Contrato para el subsistema de bitácoras, diagnóstico y auditoría (Audit Trail)
/// acorde a normativas de seguridad (ISO 27001).
/// </summary>
public interface IServicioLogging
{
    void Info(string mensaje, string? origen = null);
    void Warning(string mensaje, string? origen = null);
    void Error(string mensaje, Exception? excepcion = null, string? origen = null);
    void Auditoria(string evento, string ipOCliente, string detalles);
}

File: TcpSocketSystem_Java/src/com/tecnm/sockets/domain/puertos/IServicioLogging.java
package com.tecnm.sockets.domain.puertos;

/**
 * Puerto de auditoría y diagnóstico para el sistema de Sockets Java.
 */
public interface IServicioLogging {

    void info(String mensaje, String origen);

    void warning(String mensaje, String origen);

    void error(String mensaje, Throwable excepcion, String origen);

    void auditoria(String evento, String ipOCliente, String detalles);
}
